string id
string processed_timestamp
string base
string color
Component[] components
